### PR TITLE
sentry: reconnecting GRPC client with retries

### PR DESCRIPTION
**Pull request by battlmonstr**
* add sentry_client_test cmd to test the client API
* refactor unary_rpc to return replies
* wrap unary_rpc/streaming_rpc with reconnect/retry logic

### CODE REVIEW

File: cmd/CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +74,9 @@ if(NOT SILKWORM_CORE_ONLY)
   add_executable(sentry "sentry.cpp;${COMMON_SRC}")
   target_link_libraries(sentry PRIVATE silkworm_node silkworm-buildinfo CLI11::CLI11 silkworm_sentry)
 
+  add_executable(sentry_client_test "sentry_client_test.cpp")
+  target_link_libraries(sentry_client_test PRIVATE silkworm_sentry)
+
```
Comment (canepat) [inline cmd/CMakeLists.txt:79]: I suggest to move this `sentry_client_test` in the `cmd/test` folder where other test clients are already present (e.g. `backend_kv_test`)

File: silkworm/node/rpc/client/call.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+boost::asio::awaitable<void> reconnect_channel(grpc::Channel& channel) {
+    bool is_stopped = false;
+
+    std::function<void()> run = [&] {
+        bool is_connected = false;
+        while (!is_connected && !is_stopped && (channel.GetState(false) != GRPC_CHANNEL_SHUTDOWN)) {
+            log::Info() << "Reconnecting grpc::Channel...";
+            auto deadline = gpr_time_add(gpr_now(GPR_CLOCK_REALTIME), gpr_time_from_seconds(5, GPR_TIMESPAN));
```
Comment (canepat) [inline silkworm/node/rpc/client/call.cpp:36]: I think we could make this hard-coded 5sec a retry interval parameter that can be specified at the calling site in `unary_rpc` and `streaming_rpc`. It's not necessary to do it in this PR, though.

**Review by canepat — CHANGES_REQUESTED** [review]

File: cmd/CMakeLists.txt (hunk 1 above)
Comment (battlmonstr) [reply cmd/CMakeLists.txt]: 👍

File: silkworm/node/rpc/client/call.cpp (hunk 2 above)
Comment (battlmonstr) [reply silkworm/node/rpc/client/call.cpp:36]: sure, let's do it later; we should probably have a back-off strategy

**Review by canepat — APPROVED** [review]

**Pull request merged**